builder/build_env.py

```diff
@@ -234,19 +236,66 @@ def _create_recipes(repository, recipes, variant_config_files, variants, channel
     for recipe in config_data.get('recipes', []):
         if recipes and not recipe.get('name') in recipes:
             continue
-        packages, deps = _get_package_dependencies(recipe.get('path'),
-                                                   variant_config_files,
-                                                   variants)
+        packages, deps, versioned_deps = _get_package_dependencies(recipe.get('path'),
+                                                                   variant_config_files,
+                                                                   variants)
         output = { 'recipe' : recipe.get('name', None),
                    'repository' : repository,
                    'packages' : packages,
                    'dependencies' : deps,
+                   'versioned_dependencies' : versioned_deps,
                    'channels' : channels if channels else []}
         outputs.append(output)
 
     os.chdir(saved_working_directory)
     return outputs
 
+def _create_all_recipes(env_config_files, variants, #pylint: disable=too-many-arguments
```

Comment:
This shouldn't start with an under score, since we will be using outside of this file.